dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/ProgramObjectBundleHook.java

```diff
@@ -54,18 +71,59 @@ public void postCreate( IdentifiableObject object, ObjectBundle bundle )
     @Override
     public void postUpdate( IdentifiableObject object, ObjectBundle bundle )
     {
-        if ( !Program.class.isInstance( object ) )
+        if ( !(object instanceof Program) )
         {
             return;
         }
 
         syncSharingForEventProgram( (Program) object );
     }
 
+    @Override
+    public <T extends IdentifiableObject> List<ErrorReport> validate(T object, ObjectBundle bundle )
+    {
+        List<ErrorReport> errors = new ArrayList<>();
+
+        if ( !(object instanceof Program) )
+        {
+            return errors;
+        }
+
+        Program program = (Program) object;
+
+        if ( getProgramInstancesCount(program) != 0 )
+        {
+            errors.add(new ErrorReport( Program.class, ErrorCode.E4011, "???" )); // TODO what args should I use?
```

Comment:
You can create a new error code, and give the program that has issues, please look at ErroCode to see examples..